File: 2139-minimum-moves-to-reach-target-score/2139-minimum-moves-to-reach-target-score.cpp
class Solution {
public:
    int minMoves(int target, int maxDoubles) {
        
        int cnt = 0;
        
        while (target > 1) {
            
            if (maxDoubles == 0) {
                cnt += target - 1;
                break;
            }
            
            if (target % 2 == 0) {
                target /= 2;
                maxDoubles--;
            }
            else {
                target--;
            }
            
            cnt++;
        }
        
        return cnt;
    }
};

File: 2139-minimum-moves-to-reach-target-score/2139-minimum-moves-to-reach-target-score.py
class Solution:
    def minMoves(self, target: int, maxDoubles: int) -> int:
        
        cnt = 0
        
        while target > 1:
            
            if maxDoubles == 0:
                cnt = cnt + target - 1
                break
            
            if maxDoubles and target % 2 == 0:
                target //= 2
                maxDoubles -= 1
            else:
                target -= 1
                
            cnt += 1
                
        return cnt